src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/RemoteTypes/RemotePackageSourceProvider.cs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Configuration;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio.RemoteTypes
+{
+    public sealed class RemotePackageSourceProvider : IPackageSourceProvider, IDisposable
+    {
+        private bool _initialized = false;
+        private readonly AsyncLazyInitializer _initializer;
+        private INuGetSourceRepositoryService _sourceRepositoryService;
+        private bool _disposedValue;
+
+        public event EventHandler PackageSourcesChanged;
+
+        public string ActivePackageSourceName
+        {
+            get
+            {
+                return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
Per another comment: please do *not* use JTF.Run (or synchronously block in any other way) around calls to brokered services.